### DISCUSSION TITLE
Ethereum Single Sign-On

### DISCUSSION THREAD

**Discussion by q9f** [forem/forem#16752] [Feature Requests]
Add Ethereum Single Sign-On as an option.

Specification: 
* https://eips.ethereum.org/EIPS/eip-4361

Technical background:
* https://dev.to/q9/finally-authenticating-rails-users-with-metamask-3fj
* https://github.com/q9f/ethereum-on-rails
* https://blog.spruceid.com/sign-in-with-ethereum-ruby-library-release-and-rails-examples/
* https://github.com/spruceid/siwe-rails-examples

Relevant libraries:
* https://github.com/spruceid/siwe_rails (Local Rails Routes)
* https://github.com/spruceid/omniauth-siwe (OmniAuth Strategy)

**Comment by ildi-x** [comment]
I would also like to see this feature added. I mentioned https://login.xyz/ in this [post on DEV](https://dev.to/devteam/integrating-the-passport-with-the-forem-ecosystem-42ea) about https://passport.forem.com/
